lib/apiservers/engine/backends/cache/image_cache.go

```diff
@@ -31,34 +29,41 @@ import (
 	"github.com/docker/docker/pkg/truncindex"
 	"github.com/docker/docker/reference"
 
+	"github.com/vmware/vic/lib/apiservers/engine/backends/kv"
 	"github.com/vmware/vic/lib/apiservers/portlayer/client"
-	"github.com/vmware/vic/lib/apiservers/portlayer/client/storage"
 	"github.com/vmware/vic/lib/metadata"
-	"github.com/vmware/vic/pkg/vsphere/sys"
+	"github.com/vmware/vic/pkg/trace"
 )
 
 // ICache is an in-memory cache of image metadata. It is refreshed at startup
 // by a call to the portlayer. It is updated when new images are pulled or
 // images are deleted.
 type ICache struct {
 	m sync.RWMutex
```

Comment:
@hickeng @vburenin I implemented this pattern for both the image cache and the layer cache. Both structs are now private and use anonymous structs for (de)serialization.